python/cudf/cudf/core/column/column.py

```diff
@@ -1276,17 +1280,7 @@ def _copy_type_metadata(self: ColumnBase, other: ColumnBase) -> ColumnBase:
           and the children of `other`.
         * if none of the above, return `other` without any changes
         """
-        # TODO: This logic should probably be moved to a common nested column
-        # class.
-        if isinstance(other, type(self)):
-            if self.base_children and other.base_children:
-                base_children = tuple(
-                    self.base_children[i]._copy_type_metadata(
-                        other.base_children[i]
-                    )
-                    for i in range(len(self.base_children))
-                )
-                other.set_base_children(base_children)
+        other = other._apply_type_metadata(self.dtype)
```

Comment:
This method, along with `_copy_type_metadata_from_arrow`, is now a one-liner since all the logic is handled in `apply`. Is it worth it to keep the `copy` functions, or should we just replace all their appearances with the corresponding `apply` method call?

Reply:
If they are really the same thing I'd vote to remove the old ones, less clutter, less confusion.

Reply:
I think it's better to avoid the indirection and just delete the methods.